From 53b1143ed4dc763f90d1da9d715176581accfd17 Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <nathan@kernel.org>
Date: Wed, 10 Mar 2021 13:09:38 -0700
Subject: [PATCH 10/12] x86/vmemmap: Add __maybe_unused to
 vmemmap_pmd_is_unused

arch/x86/mm/init_64.c:835:23: error: unused function
'vmemmap_pmd_is_unused' [-Werror,-Wunused-function]
static bool __meminit vmemmap_pmd_is_unused(unsigned long addr, unsigned long end)
                      ^
1 error generated.

Signed-off-by: Nathan Chancellor <nathan@kernel.org>
---
 arch/x86/mm/init_64.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/x86/mm/init_64.c b/arch/x86/mm/init_64.c
index 0aaafa62f8fd..505b919498ee 100644
--- a/arch/x86/mm/init_64.c
+++ b/arch/x86/mm/init_64.c
@@ -849,7 +849,7 @@ static void __meminit vmemmap_flush_unused_pmd(void)
 
 #ifdef CONFIG_MEMORY_HOTPLUG
 /* Returns true if the PMD is completely unused and thus it can be freed */
-static bool __meminit vmemmap_pmd_is_unused(unsigned long addr, unsigned long end)
+static bool __maybe_unused __meminit vmemmap_pmd_is_unused(unsigned long addr, unsigned long end)
 {
 	unsigned long start = ALIGN_DOWN(addr, PMD_SIZE);
 
-- 
2.31.0

